src/services/items.js

```diff
@@ -0,0 +1,84 @@
+// I will create all the functions that I need to create, read ,update ,and delete from firestore.
+
+import {
+  collection,
+  getDocs,
+  addDoc,
+  deleteDoc,
+  doc,
+  getDoc,
+  updateDoc,
+  increment,
+} from "firebase/firestore";
+import { db } from "../firestore";
+
+// Read
+export const getAllItems = async () => {
```

Comment:
Have a look at firestore method that allows us to listen to changes to db, it would decrease the amount of state variables you have: [Firestore docs](https://firebase.google.com/docs/firestore/query-data/listen)